Add explicit asset timeframe partitions

datejada opened the pull request:
<!--
Thanks for making a pull request to TulipaEnergyModel.jl.
We have added this PR template to help you help us.

Make sure to read the contributing guidelines.

See the comments below, fill the required fields, and check the items.
-->

This PR unifies the default value of the partitions for the representatives and the timeframe. So, if no data is defined in the partition files, the default partition is `uniform` of value `1`.

## Related issues

<!-- We normally work with (i) create issue; (ii) discussion if necessary; (iii) create PR. So, at least one of the following should be true:-->

<!-- Option 1, this closes an existing issue. Fill the number below-->
Closes #1009

<!-- Option 2, this is a small fix that arguably won't need an issue. Uncomment below -->
<!--
There is no related issue.
-->

## Checklist

<!-- mark true if NA -->
<!-- leave PR as draft until all is checked -->

- [ ] I am following the [contributing guidelines](https://github.com/TulipaEnergy/TulipaEnergyModel.jl/blob/main/docs/src/90-contributing.md)

- [ ] Tests are passing
- [ ] Lint workflow is passing
- [ ] Docs were updated and workflow is passing

datejada commented:
@abelsiqueira so, from the benchmark we can see the performance problem when we have several representatives and we use the constraints over clustered year (inter-temporal constraints) by defining the timeframe partitions.

I will create a PR with the equivalent problem in version 0.10.4 to have a reference.

Also, Joaquim and Oscar recommended PProf.jl find type instabilities and bottlenecks in the code. I will give it a try on a different PR.

I will keep you posted

abelsiqueira commented on src/tmp.jl:
```diff
@@ -191,6 +191,25 @@ function tmp_create_partition_tables(connection)
+            ON asset.asset = atp.asset
+            AND asset_commission.commission_year = atp.year
+        WHERE
+            asset.is_seasonal = true
```
```suggestion
            asset.is_seasonal
```

abelsiqueira reviewed (approved):
Thanks, small simple change, pre-approved.
It is a bit concerning that this doesn't break the tests at all. It's either not affecting anything, or affecting things that don't change anything. Either way, it's another thing we have to create tests for.
Maybe we modify `Norse` to remove one of the rows in `assets-timeframe-partitions.csv`?

datejada commented:
> Thanks, small simple change, pre-approved. It is a bit concerning that this doesn't break the tests at all. It's either not affecting anything, or affecting things that don't change anything. Either way, it's another thing we have to create tests for. Maybe we modify `Norse` to remove one of the rows in `assets-timeframe-partitions.csv`?

Thanks @abelsiqueira! I have updated the Storage case since it used the `uniform` partition = 1, so it does not change anything in the case study. In addition, I have updated the docs to mention the defaults and files that are allowed to be missing, and I have also updated the concepts section as a consequence.

Once the tests pass, I will merge them; thanks!
